Add list resources by label selector method to syncer interfaces

tpantelis opened the pull request:
It's useful for users to query resources by selector filter rather than list all resources and filter afterwards.

The new method does not return an error, as `ListResources` does, to simplify usage. There are 2 error cases, neither of which is recoverable by the caller:
    
 - `cache.WaitForCacheSync` returns false. This means the informer was stopped so log a warning and return an empty list.
 - conversion from `Unstructured` to the resource type failed which is a programming error and shouldn't happen so use `r.convertNoError`. In fact, we really should just panic but that can be done in a separate PR.

The new method is called by `ListResources` which now can no longer return an error so error was removed from the return signature.

tpantelis committed:
Add list resources by label selector method to syncer interfaces

It's useful for users to query resources by selector filter
rather than list all resources and filter afterwards.

The new method does not return an error, as ListResources does, to
simplify usage. There are 2 error cases, neither of which is
recoverable by the caller:

- cache.WaitForCacheSync returns false. This means the informer was
  stopped so log a warning and return an empty list.

- conversion from Unstructured to the resource type which is a
  programming error and shouldn't happen so use r.convertNoError.
  In fact, we really should just panic but that can be done in a
  separate PR.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

skitt reviewed (approved):
You need to fix your IDE’s import settings ;-).

dfarrell07 commented on pkg/syncer/resource_syncer_test.go:
```diff
@@ -870,18 +874,108 @@ func testListResources() {
+	}
+
+	Expect(actual).To(HaveLen(len(expSpecs)))
+	for _, obj := range actual {
```
```
pkg/syncer/resource_syncer_test.go:972:2: ranges should only be cuddled with assignments used in the iteration (wsl)
	for _, obj := range actual {
	^
```
https://github.com/submariner-io/admiral/actions/runs/5600444498/jobs/10242837481?pr=684

tpantelis committed:
Remove error return from ListResources signature

The internal implementation of ListResources can no longer
return an error so modify the signature accordingly to simplify
usage.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis commented:
> You need to fix your IDE’s import settings ;-).

yeah my regular laptop is being repaired so I'm using my RH laptop and things aren't fully setup.

tpantelis commented:
@skitt @dfarrell07 I made some further changes which seems to have crossed with your reviews. Basically I removed error from the ListResources* methods to simplify usage.
